cmd/lassie/fetch.go

```diff
@@ -261,31 +261,23 @@ type progressPrinter struct {
 
 func (pp *progressPrinter) subscriber(event types.RetrievalEvent) {
 	switch ret := event.(type) {
-	case events.RetrievalEventStarted:
-		switch ret.Phase() {
-		case types.IndexerPhase:
-			fmt.Fprintf(pp.writer, "\rQuerying indexer for %s...\n", ret.PayloadCid())
-		case types.QueryPhase:
-			fmt.Fprintf(pp.writer, "\rQuerying [%s] (%s)...\n", types.Identifier(ret), ret.Code())
-		case types.RetrievalPhase:
-			fmt.Fprintf(pp.writer, "\rRetrieving from [%s] (%s)...\n", types.Identifier(ret), ret.Code())
-		}
-	case events.RetrievalEventConnected:
-		switch ret.Phase() {
-		case types.QueryPhase:
-			fmt.Fprintf(pp.writer, "\rQuerying [%s] (%s)...\n", types.Identifier(ret), ret.Code())
-		case types.RetrievalPhase:
-			fmt.Fprintf(pp.writer, "\rRetrieving from [%s] (%s)...\n", types.Identifier(ret), ret.Code())
-		}
-	case events.RetrievalEventProposed:
-		fmt.Fprintf(pp.writer, "\rRetrieving from [%s] (%s)...\n", types.Identifier(ret), ret.Code())
-	case events.RetrievalEventAccepted:
-		fmt.Fprintf(pp.writer, "\rRetrieving from [%s] (%s)...\n", types.Identifier(ret), ret.Code())
-	case events.RetrievalEventFirstByte:
-		fmt.Fprintf(pp.writer, "\rRetrieving from [%s] (%s)...\n", types.Identifier(ret), ret.Code())
-	case events.RetrievalEventCandidatesFound:
+	// case events.RetrievalEventStarted:
+	// switch ret.Phase() {
+	case events.StartedFindingCandidatesEvent:
+		fmt.Fprintf(pp.writer, "\rQuerying indexer for %s...\n", ret.PayloadCid())
+	case events.StartedRetrievalEvent:
+		fmt.Fprintf(pp.writer, "\rRetrieving from [%s] (%s)...\n", events.Identifier(ret), ret.Code())
```

Comment:
technically, you don't even need event.Identifier here no? `ret` is typed now. You can actually do .StorageProviderID() or .Protocol() if you like

Reply:
I'm torn on what's the net complexity saving though.

Reply:
Identifier() will either return the storage provider ID or "Bitswap" since bitswap events don't have an SP ID. Since we don't know if this is a bitswap event or not, I feel having Identifier() here is still useful to display the SP ID if it's there or Bitswap if it's a Bitswap event.